docs/trusted-publishers-for-all-package-repositories.md

```diff
@@ -0,0 +1,122 @@
+# Trusted Publishers for All Package Repositories
+
+Trusted Publishers is a new authentication mechanism using the OpenID Connect standard (OIDC) to exchange OIDC tokens for short-lived and tightly scoped API tokens for authenticating with package repository publishing APIs. Using short-lived API tokens removes the need to share long-lived and potentially highly privileged API tokens with external systems when publishing software. This capability is primarily beneficial for hosted, automated publishing workflows.
```

Comment:
Who is the intended audience for this document?

As a package publisher, I see the name and think "I want to be counted as trusted" but as I read this, my eyes glaze over with the jargon, making me likely to miss the note at the end.  Even if I notice it, it says "primarily" which gives just enough wiggle room that I can't tell if I should care as well (again, I want to be "trusted"!).

As someone who has maintained infrastructure and CI pipelines in the past, I feel guilted into caring about "trust" / security but the opaqueness of this introduction would have me add a significant cost multiplier to estimates, which would mean my management wouldn't sign off.

I'm assuming a security team would see this and say teams need to adopt this.  This would lead to the above reaction from infrastructure / CI people which would further widens the existing gulf between those teams that I feel like tends to happen.

Also, in both of the above roles, most of the publishing is "centralized" and this isn't as needed *but* that distinction is buried even further in the document.

For example, if the introduction said something like

> Trusted Publishers is a new authentication method that builds on the existing OpenID Connect standard (OIDC) for user infrastructure publishing to a community package repositories (e.g. CI publishing to PyPI, as opposed to maintainers publishing from their system or Homebrew's central builds).  Authentication is performed by exchanging OIDC tokens for short-lived and tightly scoped API tokens for authenticating with package repository publishing APIs. Using short-lived API tokens removes the need to share long-lived and potentially highly privileged API tokens with external systems when publishing software.

Key aspects
- This starts with what scenarios it applies to
- This assures people that this isn't completely new
- It leaves the technical summary until after that

In part, this is organizing around "bottom-line upfront" or "inverted pyramid" writing styles.

Reply:
Thank you for the suggestion! I agree that the intro is relying too much on the context of "this is for package repository maintainers because it's from this WG".

Reply:
Updated in https://github.com/ossf/wg-securing-software-repos/pull/43/commits/2fb0e2fba1ae6f76f05c0dc0f99f22cedd57dd3e